commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/CStandardTest.java

```diff
@@ -508,6 +593,146 @@ private static void assertComplex(Complex z,
+     *
+     * <pre>
+     * Odd : f(z) = -f(-z)
+     * Even: f(z) =  f(-z)
+     * </pre>
+     *
+     * <p>The results must be binary equal. This includes the sign of zero.
+     *  @param z the complex
+     * @param operation the Complex operation
+     * @param operation2 the ComplexFunctions operation
+     * @param expected the expected
+     * @param type the type
+     */
+    private static void assertComplex(Complex z,
+                                      UnaryOperator<Complex> operation,
+                                      ComplexUnaryOperator<ComplexNumber> operation2,
+                                      Complex expected,
+                                      FunctionType type) {
+        assertComplex(z, operation, operation2, expected, type, UnspecifiedSign.NONE);
+    }
+
+    /**
+     * Assert the operation on the complex number is equal to the expected value. If the
+     * imaginary part is not NaN the operation must also satisfy the conjugate equality.
+     *
+     * <pre>
+     * op(conj(z)) = conj(op(z))
+     * </pre>
+     *
+     * <p>If the function type is ODD/EVEN the operation must satisfy the function type equality.
+     *
+     * <pre>
+     * Odd : f(z) = -f(-z)
+     * Even: f(z) =  f(-z)
+     * </pre>
+     *
+     * <p>An ODD/EVEN function is also tested that the conjugate equalities hold with {@code -z}.
+     * This effectively enumerates testing: (re, im); (re, -im); (-re, -im); and (-re, im).
+     *
+     * <p>The results must be binary equal; the sign of the complex number is first processed
+     * using the provided sign specification.
+     *
+     * @param z the complex
+     * @param operation the Complex operation
+     * @param operation2 the ComplexFunctions operation
+     * @param expected the expected
+     * @param type the type
+     * @param sign the sign specification
+     */
+    private static void assertComplex(Complex z,
+                                      UnaryOperator<Complex> operation, ComplexUnaryOperator<ComplexNumber> operation2, Complex expected,
+                                      FunctionType type, UnspecifiedSign sign) {
+        // Developer note: Set the sign specification to UnspecifiedSign.NONE
+        // to see which equalities fail. They should be for input defined
+        // in ISO C99 with an unspecified output sign, e.g.
+        // sign = UnspecifiedSign.NONE
+
+        // Test the operation
+        final Complex c = operation.apply(z);
+        final ComplexNumber c1 = operation2.apply(z.getReal(), z.getImaginary(), ComplexNumber::new);
+        final Complex t1 = sign.removeSign(c);
+        final Complex t2 = sign.removeSign(expected);
+        if (!equals(t1.getReal(), t2.getReal()) ||
+            !equals(t1.getImaginary(), t2.getImaginary())) {
+            Assertions.fail(
+                String.format("Operation failed (z=%s). Expected: %s but was: %s (Unspecified sign = %s)",
```

Comment:
This formatting message includes c1 as an argument but the string format does not require it.